internal/uvm/memory.go

```diff
@@ -0,0 +1,45 @@
+package uvm
+
+import (
+	"context"
+	"fmt"
+
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+const (
+	bytesPerPage = 4096
+	bytesPerMB   = 1024 * 1024
+)
+
+// UpdateMemory makes a call to the VM's orchestrator to update the VM's size in MB
+func (uvm *UtilityVM) UpdateMemory(ctx context.Context, sizeInMB uint64) error {
+	req := &hcsschema.ModifySettingRequest{
+		ResourcePath: memoryResourcePath,
+		Settings:     sizeInMB,
+	}
+	if err := uvm.modify(ctx, req); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetAssignedMemoryInMB returns the amount of assigned memory for the UVM in MBs
+func (uvm *UtilityVM) GetAssignedMemoryInMB(ctx context.Context) (uint64, error) {
+	props, err := uvm.hcsSystem.PropertiesV2(ctx, hcsschema.PTMemory)
+	if err != nil {
+		return 0, err
+	}
+	if props.Memory == nil {
+		return 0, fmt.Errorf("no memory properties returned for system %s", uvm.id)
+	}
+	if props.Memory.VirtualMachineMemory == nil {
+		return 0, fmt.Errorf("no virtual memory properties returned for system %s", uvm.id)
+	}
```

Comment:
Curious on if you've ever ran into either of these being nil, or this is just to be safe. We do the first check here but just use VirtualMachineMemory without doing a check right after so if you've seen this returned as nil we should update. https://github.com/microsoft/hcsshim/blob/1a9588fd692bcf7e7e8df6c0344e1f04cc5a2544/internal/uvm/stats.go#L124

Reply:
This is mostly to be safe, but I'm not familiar with the internal implementation, so I'm not sure if there are cases where it could be nil.

Reply:
Gotcha